.github/workflows/docker.yml

```diff
@@ -18,68 +22,78 @@
+
+      - uses: docker/setup-qemu-action@v2
+
+      - uses: docker/metadata-action@v5
+        id: meta
+        with:
+          images: ${{ env.DOCKER_IMAGE }}
+          flavor: |
+            prefix=heighliner-,onlatest=true
+          tags: |
+            type=edge
+            type=ref,enable=${{ github.event_name == 'pull_request' }},event=branch
+            type=ref,event=tag
+
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+        # - uses: strangelove-ventures/heighliner-build-action@v1.0.1
+      - uses: strangelove-ventures/heighliner-build-action@main
         with:
-          fetch-depth: 0
+          heighliner-tag: v1.5.5
+          chain: provenance
+          local: true
+          tag: ${{ steps.meta.outputs.version }}
+          registry: provenanceio
+          additional-args: --alpine-version=3.18
+          skip: ${{ github.event_name == 'pull_request' }}
 
-      - name: Setup go
-        uses: actions/setup-go@v5
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
         with:
           go-version: '1.21'
 
-      - name: Go mod vendor
-        run: |
-         go mod vendor
+      - run: go mod vendor
 
-      - name: Prepare
-        id: prep
-        run: |
-          DOCKER_IMAGE=provenanceio/provenance
-          VERSION=noop
-          if [[ $GITHUB_REF == refs/tags/* ]]; then
-            VERSION=${GITHUB_REF#refs/tags/}
-          elif [[ $GITHUB_REF == refs/heads/* ]]; then
-            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
-            if [ "${{ github.event.repository.default_branch }}" = "$VERSION" ]; then
-              VERSION=latest
-            fi
-          fi
-          TAGS="${DOCKER_IMAGE}:${VERSION}"
-          if [[ $VERSION =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
-            TAGS="$TAGS,${DOCKER_IMAGE}:${VERSION}"
-          fi
-          echo "Setting output: version=${VERSION}"
-          echo "version=${VERSION}" >> "$GITHUB_OUTPUT"
-          echo "Setting output: tags=${TAGS}"
-          echo "tags=${TAGS}" >> "$GITHUB_OUTPUT"
-          created="$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
-          echo "Setting output: created=$created"
-          echo "created=$created" >> "$GITHUB_OUTPUT"
+      - uses: docker/setup-buildx-action@v3
 
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v3
+      - uses: docker/setup-qemu-action@v2
 
-      - name: Available platforms
-        run: echo ${{ steps.buildx.outputs.platforms }}
+      - uses: docker/metadata-action@v5
+        id: meta
+        with:
+          images: ${{ env.DOCKER_IMAGE }}
+          tags: |
+            type=edge
+            type=ref,enable=${{ github.event_name == 'pull_request' }},event=branch
+            type=ref,event=tag
```

Comment:
Stop suggesting that we use secrets for everything. Secrets are only useful for a very limited scope of values related to information that should never be made public, and that no one should know the actual value of. Using secrets for anything else significantly reduces flexibility and maintainability.

Reply:
Seriously. You're like a kid that can't remember two sentences in a row.

In this line, we are setting the `images` value to be the `DOCKER_IMAGE` environment variable.

Stop suggesting that we change things to the way they already are.

Reply:
Maybe you need to familiarize yourself with GitHub workflow syntax. You can tell that's what these are because they're in the `.github/workflows` directory as specified by GitHub. Maybe that's why you keep making completely misguided suggestions in here.